barretenberg/cpp/src/barretenberg/proof_system/circuit_builder/generated/AvmMini_trace.cpp

```diff
@@ -0,0 +1,346 @@
+#include <array>
+#include <cassert>
+#include <cstddef>
+#include <cstdint>
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <sys/types.h>
+#include <vector>
+
+#include "./AvmMini_trace.hpp"
+
+#include "barretenberg/relations/generated/AvmMini.hpp"
+namespace proof_system {
+
+void AvmMiniTraceBuilder::reset()
+{
+    mainTrace.clear();
+    memTrace.clear();
+    ffMemory.fill(FF(0));
+}
+
+bool AvmMiniTraceBuilder::compareMemEntries(const MemoryTraceEntry& left, const MemoryTraceEntry& right)
+{
+    if (left.m_addr < right.m_addr) {
+        return true;
+    }
+
+    if (left.m_addr > right.m_addr) {
+        return false;
+    }
+
+    if (left.m_clk < right.m_clk) {
+        return true;
+    }
+
+    if (left.m_clk > right.m_clk) {
+        return false;
+    }
+
+    // No safeguard in case they are equal. The caller should ensure this property.
+    // Otherwise, relation will not be satisfied.
+    return left.m_sub_clk < right.m_sub_clk;
+}
+
+void AvmMiniTraceBuilder::insertInMemTrace(uint32_t m_clk, uint32_t m_sub_clk, uint32_t m_addr, FF m_val, bool m_rw)
+{
+    auto newMemEntry = MemoryTraceEntry{
+        .m_clk = m_clk,
+        .m_sub_clk = m_sub_clk,
+        .m_addr = m_addr,
+        .m_val = m_val,
+        .m_rw = m_rw,
+    };
+
+    auto insertionIndex = std::lower_bound(memTrace.begin(), memTrace.end(), newMemEntry, compareMemEntries);
+    memTrace.insert(insertionIndex, newMemEntry);
```

Comment:
insert is O(N), so reusing this operation will become O(N^2). It's better to use sort in the end